%\textsl{}%!TEX TS-options = --shell-escape
%!TEX TS-program = pdflatex
\documentclass[%
   10pt,              % Schriftgroesse
   nenglish,           % wird an andere Pakete weitergereicht
   a4paper,           % Seitengroesse
   DIV11,             % Textbereichsgroesse (siehe Koma Skript Dokumentation !)
]{scrartcl}%     Klassen: scrartcl, scrreprt, scrbook, article
% -------------------------------------------------------------------------

\usepackage[utf8]{inputenc} % Font Encoding, benoetigt fuer Umlaute
\usepackage[english]{babel}   % \textsl{}Spracheinstellung

\usepackage[T1]{fontenc} % T1 Schrift Encoding
\usepackage{textcomp}    % Zusatzliche Symbole (Text Companion font extension)
\usepackage{lmodern,dsfont}     % Latin Modern Schrift
\usepackage{dsfont}
\usepackage{color}
%\usepackage{wasysym}
\usepackage{ulem}
\usepackage{graphicx}
\usepackage{grffile} %allows to use pngs
\usepackage{eurosym}
%\usepackage{txfonts}
\usepackage{stmaryrd}
\usepackage{amsfonts}
\usepackage{amsmath}
\usepackage{hyperref}
\usepackage{tikz}
\usepackage{multirow}
\usepackage{listings}
\usepackage{etextools}
\usepackage{ifthen}
%\usepackage{TikZ} %phylogenetischer Baum
%\usetikzlibrary{calc, shapes, backgrounds} %für die Phylogenetische bäume
%\usetikzlibrary{automata,arrows}
\usepackage{subfigure} 


% Definition des Headers
\usepackage{geometry}
\geometry{a4paper, top=3cm, left=3cm, right=3cm, bottom=3cm, headsep=0mm, footskip=0mm}
\renewcommand{\baselinestretch}{1.3}\normalsize

\def\header#1#2#3#4#5#6#7{\pagestyle{empty}
\noindent
\begin{minipage}[t]{0.6\textwidth}
\begin{flushleft}
\textbf{#4}\\% Fach
#6\\% Semester
Tutor: #2  % Tutor 
\end{flushleft}
\end{minipage}
\begin{minipage}[t]{0.4\textwidth}
\begin{flushright}
\points{#7}% Punktetabelle
\vspace*{0.2cm}
#5%  Names
\end{flushright}
\end{minipage}

\begin{center}
{\Large\textbf{ Assignment #1}} % Blatt

{(Abgabe am #3)} % Abgabedatum
\end{center}
}

\newenvironment{vartab}[1]
{
    \begin{tabular}{ |c@{} *{#1}{c|} } %\hline
}{
    \end{tabular}
}

\newcommand{\myformat}[1]{& #1}

\newcommand{\entry}[1]{
  \edef\result{\csvloop[\myformat]{#1}}
  \result \\ \hline
}

\newcommand{\numbers}[1]{
  \newcounter{ctra}
\setcounter{ctra}{1}
\whiledo {\value{ctra} < #1}%
{%
  \myformat{\thectra}
  \stepcounter{ctra}%
}
\myformat{\thectra}
}
\newcommand{\emptyLine}[1]{
  \newcounter{ctra1}
\setcounter{ctra}{1}
\whiledo {\value{ctra1} < #1}%
{%
  \myformat{\hspace*{0.5cm}}
  \stepcounter{ctra1}%
}
}

\newcommand{\points}[1]{
\newcounter{colmns}
\setcounter{colmns}{#1}
\stepcounter{colmns}
  \begin{vartab}{\thecolmns}
    \numbers{#1} & $\sum$\\\hline
    \emptyLine{\thecolmns}\\
  \end{vartab}
}


\begin{document}
%\header{Blatt}{Tutor}{Abgabedatum}{Vorlesung}{Bearbeiter}{Semester}{Anzahl Aufgaben}
\header{4}{Alexander Seitz}{9. November 2015}{Bioinformatics I}{\\Jonas Ditz \\\& Benjamin Schroeder}{WS 15/16}{3}

\section*{Theoretical Assignment - \textsl{Optimal multiple alignment}}

To calculate our MSA we use the recursion written down on page 50 in the script. 

\begin{equation}
 F(i_1, i_2, i_3) = max \begin{cases}
                         F(i_1 -1, i_2 -1, i_3 -1) + s_{SP}(a_{1i_1},a_{2i_2},a_{3i_3}) \\
                         \\
                         F(i_1 -1, i_2 -1, i_3) + s_{SP}(a_{1i_1},a_{2i_2},-) \\
                         F(i_1 -1, i_2, i_3 -1) + s_{SP}(a_{1i_1},-,a_{3i_3}) \\
                         F(i_1, i_2 -1, i_3 -1) + s_{SP}(-,a_{2i_2},a_{3i_3}) \\
                         \\
                         F(i_1 -1, i_2, i_3) + s_{SP}(a_{1i_1},-,-) \\
                         F(i_1, i_2 -1, i_3) + s_{SP}(-,a_{2i_2},-) \\
                         F(i_1, i_2, i_3 -1) + s_{SP}(-,-,a_{3i_3}) \\
                        \end{cases} \nonumber
\end{equation}

\noindent Such a recursion results in a three-dimensional matrix. Since such a matrix is really difficult to 
sketch, we split it into three two-dimensional matrices.  

\begin{figure}[ht]
\centering
\subfigure{
\begin{tabular}{c|cccc}
 0 & 0 & C & C & T \\
 \hline
 0 & \textcolor{red}{\textbf{0}} & -2 & -4 & -6 \\
 C & -2 & \textcolor{red}{\textbf{-2}} & -4 & -10 \\
 T & -4 & -8 & -8 & -6 \\
 T & -6 & -10 & -12 & -10 \\
\end{tabular}}
\subfigure{
\begin{tabular}{c|cccc}
 T & 0 & C & C & T \\
 \hline
 0 & -2 & -4 & -6 & -8 \\
 C & -4 & -4 & -6 & -10 \\
 T & -6 & -8 & \textcolor{red}{\textbf{-4}} & 2 \\
 T & -8 & -10 & -8 & 2 \\
\end{tabular}}
\subfigure{
\begin{tabular}{c|cccc}
 C & 0 & C & C & T \\
 \hline
 0 & -4 & -6 & -8 & -10 \\
 C & -6 & 2 & 2 & -8 \\
 T & -8 & -6 & -2 & 0 \\
 T & -10 & -8 & -6 & \textcolor{red}{\textbf{-6}} \\
\end{tabular}}
\caption[DP matrix of MSA]{DP matrix of MSA, traceback is shown in red}
\end{figure}

\noindent If we fill the DP matrix using this recursion, we get several optimal alignments. One of 
them is the following:

\begin{equation}
 \begin{pmatrix}
  C & T & T \\
  - & T & C \\
  C & C & T
 \end{pmatrix}
\end{equation}

\noindent with score $\alpha_{SP}(A^*) = S(A,B) + S(A,C) + S(B,C) = -2 + 2 + (-6) = -6$.

\section*{Theoretical Assignment - \textsl{Progressive alignment}}
Progressive alignment methodes are one way to create multiple sequence alignments (MSA), in this task we had to accomplish manualy a progressive alignment. The first step for the progressive alignment was to generate the pairwise global distance matrices. Afterwards the global optima were entered into a new table, which was used to generate the guide tree in the second step.(figure \ref{DP2}).
\begin{figure}[h]
	\includegraphics[width=\textwidth]{Img/Exercise2-DP-Matrix.png}
	\caption{ Global alignments lead to the basis table used to generate the guide tree}
	\label{DP2}
\end{figure}

In step 2 the guide tree was created by the UPGMA methode in 3 substeps. After every step, which added a new cluster, the distance table was updated. The result from this step was the guide tree, which was used to apply the 'complete alignment' methode in the last step, the progressive alignment step.(figure \ref{guidetree2})


\begin{figure}[h]
	\includegraphics[width=\textwidth]{Img/Exercise2-Guidetree.png}
	\caption{ The guide tree is made from the final table from step 1}
	\label{guidetree2}
\end{figure}

The last step was creating the actual MSA. The 'complete alignment' methode was used. This means after every step, the distance of the new sequences was determined to each other sequence in the cluster. The optimal resulting score was used to get the best alignment for the new Sequence. after 3 steps the MSA was completed. (figure \ref{completealign})
\begin{figure}[h]
	\includegraphics[width=\textwidth]{Img/Exercise2-CompleteAlignment.png}
	\caption{ Complete alignment methode is used to get an MSA from the guidetree from step 2	}
	\label{completealign}
\end{figure}
\section*{Practical Assignment - \textsl{Comparing multiple alignment}}
 
\end{document}